google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -574,6 +574,23 @@ public PooledSessionFuture replaceSession(
     }
   }
 
+  class MultiplexedSessionReplacementHandler
+      implements SessionReplacementHandler<MultiplexedSessionFuture> {
+    @Override
+    public MultiplexedSessionFuture replaceSession(
+        SessionNotFoundException e, MultiplexedSessionFuture session) {
+      /**
+       * For multiplexed sessions, we would never obtain a {@link SessionNotFoundException}. Hence,
+       * this method will ideally never be invoked.
+       */
+      logger.log(
+          Level.WARNING,
+          String.format(
+              "Replace session invoked for multiplexed " + "session => %s", session.getName()));
+      return session;
```

Comment:
I think that we should just propagate the `SessionNotFoundException` here. It is not something that should happen, but if it does happen for whatever reason, the current implementation is likely to just get into an eternal loop.

Reply:
Sure, I will make this change to propagate the exception.